### ISSUE TITLE
Add typescript support

### ISSUE THREAD

**Issue by punisher97** [carloscuesta/react-native-error-boundary#11]
Hi, please add typescript support, create index.d.ts file and set key 'types' in package.json.

Thanks
```typescript
declare module 'react-native-error-boundary' {
  import React from 'react';

  export type ErrorBoundaryProps = {
    children: React.ReactNode;
    onError?: (error: Error, stackTrace: string) => void;
    FallbackComponent?: React.ComponentType<{ error: Error; resetError: () => void }>;
  };

  export default class ErrorBoundary extends React.Component<ErrorBoundaryProps> {}
}
```

**Comment by carloscuesta** [comment]
Hi @punisher97 I'm not quite used to TypeScript,

This are the docs right https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html that point how to declare module types

**Comment by carloscuesta** [comment]
Can you take a look at the pr ? @punisher97

**Comment by carloscuesta** [comment]
Released at `1.1.0` 🎉
